### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Rhaegar121**
In this PR, I have completed the following tasks.

1. Created `Classroom` class with `label` instance variable
2. Created relationships between classroom and student
3. Created `Book` class with `title` and `author` instance variables
4. Created `Rental` class with `date` instance variable
5. Created relationships between `Person` and `Book` using intermediate class `Rental`

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require_relative 'nameable'
```
Comment (tufoinnkuo10) [inline person.rb:1]: - [ ] kindly remove the nameable as there is no class here with it in this milestone

File: person.rb [hunk 2]
```diff
@@ -10,6 +10,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (tufoinnkuo10) [inline person.rb:13]: - [ ] like i mentioned above kindly add a method for rental here as well so you can achieve the 'has-many' association
eg 
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (tufoinnkuo10) [inline book.rb:9]: - [ ] nice work so far. please ensure A method for adding rentals should be present. see more [here](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars`)
- please kindly ensure `the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association`
eg 
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Rhaegar121 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ Documented your work professionally
✅ Associations well created

Required Changes ♻️
Check the comments under the review

tfufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: person.rb (hunk 1 above)
Comment (Rhaegar121) [reply person.rb:1]: But if I remove require_relative I think there will be rubocop errors.

**Review by omarsalem7 — APPROVED** [review]
Hi @Rhaegar121 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**